_posts/2023-01-12-gsoc-simultaneous-trajectory-execution.md

```diff
@@ -0,0 +1,48 @@
+---
+author: Cristian C. Beltran-Hernandez
+comments: false
+date: 2023-01-12
+layout: post
+title: Simultaneous Trajectory Execution
+media_type: image
+media_link: /assets/images/blog_posts/gsoc-2022-badge.svg
+description: Google Summer of Code 2022 project's result
+categories:
+- MoveIt
+- Google
+---
+
+This post summarizes my contributions during the [Google Summer of Code 2022](https://summerofcode.withgoogle.com/programs/2022/projects/mpusZVc2) where I have extended the trajectory execution capabilities of MoveIt. 
+
+# Motivation
+
+Before this project, MoveIt only allowed the execution of trajectories in strict sequential order, which limited the options to simultaneously control two or more robots to only synchronized motions.
+
+The goal was then to allow the simultaneous execution of multiple trajectories on systems with multiple robots.  For example, in a dual-arm robotic system, each arm can execute a different set of trajectories without needing to wait for the other arm to finish moving (sequential execution) or manually synchronizing the motion of both arms into a single trajectory. To preserve collision-free guarantees an extra collision check needs to be performed right before the execution of new trajectories to prevent collisions with active trajectories. 
+
+## Example Use Cases
+1. Several trajectories are planned and executed through the `MoveIt Motion Planning Rviz` plugin. 
+![simultaneous-execution-rviz](https://user-images.githubusercontent.com/3798796/196121126-6842200c-6a81-42cc-bc7a-b9f7200d7a2b.gif)
```

Comment:
Shouldn't the GIFs be added to the web site directly? It seems like content under https://user-images.githubusercontent.com/ could disappear in the future.